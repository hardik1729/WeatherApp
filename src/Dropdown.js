import React from "react";
import "./style.css";

class Dropdown extends React.Component {
  constructor() {
    super();
    this.state = {
      displayMenu: false,
      name: "Location"
    };
    this.showDropdownMenu = this.showDropdownMenu.bind(this);
    this.hideDropdownMenu = this.hideDropdownMenu.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }
  showDropdownMenu(event) {
    this.setState({ displayMenu: true }, () => {
      document.addEventListener('click', this.hideDropdownMenu);
    });
  }
  hideDropdownMenu() {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener('click', this.hideDropdownMenu)});
  }

  handleClick(event){
    //console.log("from handleclick "+event.currentTarget.dataset.id);
    this.setState({
      name: event.currentTarget.dataset.id
    });
    this.props.selectionDidChange(event.currentTarget.dataset.id);
  };
  render() {
    return (
      <div className="dropdown" style={{ background: "#61dafb", width: "200px", position: "left"}}>
        <div className="button" onClick={this.showDropdownMenu}>
          {this.state.name}
        </div>
        {this.state.displayMenu ? (
          <ul>
            <li data-id="Liverpool" onClick={this.handleClick}>
              Liverpool
            </li>
            <li data-id="Bristol" onClick={this.handleClick}>
              Bristol
            </li>
            <li data-id="Manchester" onClick={this.handleClick}>
              Manchester
            </li>
          </ul>
        ) : null}
      </div>
    );
  }
}
export default Dropdown;
