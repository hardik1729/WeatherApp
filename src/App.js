import React from 'react';
import logo from './logo.svg';
import './App.css';
import Dropdown from './Dropdown'

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      selected: props.selected,
      data: [],
    }

    this.selectionDidChange = this.selectionDidChange.bind(this);
  }

  selectionDidChange (newValue) {
    let data=fetch("http://open.mapquestapi.com/directions/v2/alternateroutes?key=86e3wu3G1SG1k22vRkXsygIwUE8wh6zU&from=mumbai&to=Indore")
    .then(response => {response.json()})
    this.setState({
      selected: newValue,
      data: data,
    });
  }

  render () {
    return (
      <div className="App">
        <Dropdown selectionDidChange={this.selectionDidChange}/>
        <header className="App-header">
          {
            this.state.selected ? (
              <p>
                {this.state.data}
              </p>  
            ) :
            (
              <div>
                <img src={logo} className="App-logo" alt="logo" />
                <p>
                  Weather Details to be displayed here
                </p>
              </div> 
            )
          }
        </header>
      </div>
    );
  }
}

export default App;