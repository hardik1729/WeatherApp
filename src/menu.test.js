import React from 'react';
import TestRenderer from 'react-test-renderer';
import Dropdown from './Dropdown';
import App from './App';

it('it renders with displaymenu', () => {
    const dropdown = TestRenderer.create(<Dropdown selectionDidChange={jest.fn()}/>);
    dropdown.root.instance.showDropdownMenu();
    expect(dropdown.toJSON()).toMatchSnapshot();
});

it('it renders without displaymenu', () => {
    const dropdown = TestRenderer.create(<Dropdown selectionDidChange={jest.fn()}/>);
    dropdown.root.instance.hideDropdownMenu();
    expect(dropdown.toJSON()).toMatchSnapshot();
});

it('Calls selectionDidChange function', () => {
    const dropdown = TestRenderer.create(<Dropdown selectionDidChange={jest.fn()}/>);
    let places = [
        {place:"Liverpool",distance:100},
        {place:"Bristol",distance:200},
        {place:"Manchester",distance:300}
    ];
    places.forEach(element => {
        let event={
                    currentTarget:{
                        dataset:{
                            id: element.place
                        }
                    }
                };
        dropdown.root.instance.handleClick(event);
        expect(dropdown.toJSON()).toMatchSnapshot();
    });
        
});