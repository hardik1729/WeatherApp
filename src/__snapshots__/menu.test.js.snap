// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calls selectionDidChange function 1`] = `
<div
  className="dropdown"
  style={
    Object {
      "background": "#61dafb",
      "position": "left",
      "width": "200px",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    Liverpool
  </div>
</div>
`;

exports[`Calls selectionDidChange function 2`] = `
<div
  className="dropdown"
  style={
    Object {
      "background": "#61dafb",
      "position": "left",
      "width": "200px",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    Bristol
  </div>
</div>
`;

exports[`Calls selectionDidChange function 3`] = `
<div
  className="dropdown"
  style={
    Object {
      "background": "#61dafb",
      "position": "left",
      "width": "200px",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    Manchester
  </div>
</div>
`;

exports[`it renders with displaymenu 1`] = `
<div
  className="dropdown"
  style={
    Object {
      "background": "#61dafb",
      "position": "left",
      "width": "200px",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    Location
  </div>
  <ul>
    <li
      data-id="Liverpool"
      onClick={[Function]}
    >
      Liverpool
    </li>
    <li
      data-id="Bristol"
      onClick={[Function]}
    >
      Bristol
    </li>
    <li
      data-id="Manchester"
      onClick={[Function]}
    >
      Manchester
    </li>
  </ul>
</div>
`;

exports[`it renders without displaymenu 1`] = `
<div
  className="dropdown"
  style={
    Object {
      "background": "#61dafb",
      "position": "left",
      "width": "200px",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    Location
  </div>
</div>
`;
